tests/cli-misc.rs

```diff
@@ -122,6 +122,17 @@ fn update_works_without_term() {
     });
 }
 
+// Issue #1738
+#[test]
+fn show_works_with_dumb_term() {
```

Comment:
Unfortunately I'm not convinced this works anyway because the "output" is not a terminal when running under our test setup, it's a pipe.

Reply:
The test is failed before you patch. It was weird that if I use `cmd.output()`, the test is passed.
But with `cmd.spawn` the test is failed.

Reply:
Aah so .spawn() is leaving stdout/stderr going to the terminal.  Do you see output from rustup show or is that still swallowed somehow?  Also was it failing in CI beforehand?  I'm not entirely sure if travis et al run their tests in a pseudoterminal or not.

Reply:
From the docs, it tells:
- `spawn`: Executes the command as a child process, returning a handle to it.
By default, stdin, stdout and stderr are inherited from the parent.
- `output`: Executes the command as a child process, waiting for it to finish and collecting all of its output.
By default, stdout and stderr are _captured_ (and used to provide the resulting output).

Reply:
Yes, I'd just not spotted the use of `.spawn()` rather than `.output()`.  What I'm concerned about is that this test might do nothing in Travis/Appveyor if they don't use pseudoterminals for builds (I'm not sure if they do or not)  Do they?

Reply:
Given the colour I'm going to guess they might, *BUT* this ends up with https://travis-ci.com/rust-lang/rustup.rs/builds/107899773#L5995 type test output spew which isn't great.

I guess I'm not sure if I think the messy output of the tests is worth it for adding this test, certainly I think the complexity of a cross-platform pseudoterminal library dependency addition might be too much though :(

Reply:
Here the link that the test is failed on Travis before your patch: https://travis-ci.com/rust-lang/rustup.rs/builds/107833119#L6000

Reply:
OK, I accept that Travis must be testing it with a PTY, that's good.

I'm still not sure it's worth the test spew but I suppose it's very small spew so for now we'll accept it.